### PR TITLE
[Misc] Add unit test for merge_attn_states kernel

### PR DESCRIPTION
**Pull request by pmanczak**
## Purpose

Adds a device-agnostic unit test for the `merge_attn_states` Triton kernel
(Attention group) in `tests/kernels/attention/test_merge_attn_states_torch_ref.py`.

The existing `test_merge_attn_states.py` compares the Triton kernel against a
custom CUDA op (`merge_attn_states_cuda`) and skips on non-CUDA platforms, so it
cannot run on XPU. This test instead uses a naive PyTorch reference as the oracle
(Section 2.2 of arxiv 2501.01005 — the softmax-weighted combination of two
split-KV partial attention results), making it runnable on both CUDA and XPU
(`current_platform.device_type`, no skip guard). Kept as a separate file so the
existing CUDA-parity test is preserved.

Covers output and output_lse across token counts, head counts, head sizes, and
fp16/bf16, including ~10% +inf lse entries (a split that attended to no keys).

## Test Plan

`python -m pytest tests/kernels/attention/test_merge_attn_states_torch_ref.py -v`

## Test Result

All 36 cases pass.

### CODE REVIEW

**Review by rogerxfeng8 — APPROVED** [review]
The code itself looks ok. However, there is tests/kernels/attention/test_merge_attn_states.py already. We are expected to follow the same test logic there, for xpu path, it needs to compare to the xpu version merge_attn_states_kernel which we have in vllm-xpu-kernels.

**Review by jikunshang — CHANGES_REQUESTED** [review]
I don't think this is valid change. for xpu, we should fix  `test_merge_attn_states.py` instead.

**Comment by rogerxfeng8** [comment]
> I don't think this is valid change. for xpu, we should fix `test_merge_attn_states.py` instead.

Agree. the fix should go to original test script.

**Review by rogerxfeng8 — CHANGES_REQUESTED** [review]
Should work on original test script.

**Comment by jikunshang** [comment]
https://github.com/vllm-project/vllm-xpu-kernels/pull/467 I think we can simplify a lot once this launch
